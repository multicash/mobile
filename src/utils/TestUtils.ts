import fs from 'fs'
import path from 'path'
import { compileVueToRn } from 'vue-native-scripts'

export const getTestComponent = (component: string): any => {
  const data = fs.readFileSync(path.join(__dirname, '..', component), 'utf8')

  return compileVueToRn(data).output
}
