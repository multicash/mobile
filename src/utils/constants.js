export default {
  priceApi: 'https://api.vergecurrency.network/price/api/v1/price',
  ipApi: 'https://api.vergecurrency.network/price/api/v1/ip/',
  vwsApi: 'https://api.vergecurrency.network/vws/api',
  bnApi: 'https://api.vergecurrency.network/node/api/XVG',
  explorer: 'https://verge-blockchain.info',
  termsOfUse: 'https://vergecurrency.com/wallets/terms',
  feePerKb: 100000,
  satoshiDivider: 1000000,
  decimalPerSatoshi: 6,
  defaultCurrencyCode: 'USD',
  paperKeyLength: 12
}
