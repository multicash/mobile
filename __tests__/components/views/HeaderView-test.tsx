import React from 'react'
import renderer from 'react-test-renderer'
import { getTestComponent } from '@/utils/TestUtils'

const HeaderView = getTestComponent('components/views/HeaderView.vue')

describe('Render header view component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<HeaderView />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
