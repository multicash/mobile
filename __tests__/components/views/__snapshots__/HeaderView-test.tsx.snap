// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render header view component renders correctly 1`] = `
<import __react__vue__Vue, { observer as __react__vue__observer } from 'vue-native-core'
import __react__vue__ReactNative from 'react-native'
import __react__vue__PropType from 'prop-types'
import { buildNativeComponent as __react__vue__buildNativeComponent } from 'vue-native-helper'
import { bindNativeClass as __react__vue__bindClass,bindNativeStyle as __react__vue__bindStyle,mergeNativeStyleAndNativeClass as __react__vue__mergeNativeStyleAndNativeClass,mergeProps as __react__vue__mergeProps,_toString as __react__vue__toString } from 'vue-native-helper'
import {
      createElement as __react__vue__createElement,
      Component as __react__vue__Component
    } from 'react'

import { pageSubtitle, pageTitle } from '@/core/support/styles'
const __react__vue__options =  {
  name: 'HeaderView',

  props: {
    title: {
      type: String,
      required: true
    },
    subtitle: {
      type: String
    },
    imageBackground: {
      default: null
    },
    imageForeground: {
      default: null
    },
    image: {
      default: null
    }
  },

  computed: {
    styles () {
      return stylesStore(this.isDarkScheme, !!this.imageForeground)
    }
  }
}

const stylesStore = (isDarkScheme, hasImageForeground) => {
  return {
    headerContainer: {
      marginHorizontal: 20,
      marginTop: 20,
      justifyContent: 'center',
      alignItems: 'center'
    },

    headerTitle: {
      ...pageTitle(isDarkScheme)
    },

    headerSubtitle: {
      ...pageSubtitle(isDarkScheme)
    },

    image: {
      width: 60,
      height: 60,
      resizeMode: 'contain'
    },

    headerImage1: {
      zIndex: 10,
      width: 40,
      height: 40,
      marginTop: 0,
      marginRight: hasImageForeground ? -25 : 0,
      resizeMode: 'contain'
    },

    headerImage2: {
      width: 60,
      height: 60,
      resizeMode: 'contain',
      marginTop: 10,
      marginBottom: 2.5
    }
  }
}

const __react__vue__render = function render(vm) {
    return __react__vue__createElement(vm.$options.components['View'], __react__vue__mergeProps.call(this, this.props.__react__vue__nativeEvents, {
        ref: ref => {
            this.setRootRef(ref);
            this.props['__react__vue__setRef'] && this.props['__react__vue__setRef'](ref);
        },
        style: __react__vue__mergeNativeStyleAndNativeClass(__react__vue__bindClass.call(this, {
            parentClass: this.props.style
        }), __react__vue__bindStyle(vm['styles'].headerContainer, undefined, undefined))
    }), __react__vue__createElement(vm.$options.components['View'], {
        style: __react__vue__mergeNativeStyleAndNativeClass(undefined, __react__vue__bindStyle({
            flexDirection: 'row'
        }, undefined, undefined))
    }, vm['imageForeground'] ? __react__vue__createElement(vm.$options.components['Image'], {
        source: vm['imageForeground'],
        style: __react__vue__mergeNativeStyleAndNativeClass(undefined, __react__vue__bindStyle(vm['styles'].headerImage1, undefined, undefined))
    }) : null, vm['imageBackground'] ? __react__vue__createElement(vm.$options.components['Image'], {
        source: vm['imageBackground'],
        style: __react__vue__mergeNativeStyleAndNativeClass(undefined, __react__vue__bindStyle(vm['styles'].headerImage2, undefined, undefined))
    }) : vm['image'] ? __react__vue__createElement(vm.$options.components['Image'], {
        source: vm['image'],
        style: __react__vue__mergeNativeStyleAndNativeClass(undefined, __react__vue__bindStyle(vm['styles'].image, undefined, undefined))
    }) : null), __react__vue__createElement(vm.$options.components['Text'], {
        style: __react__vue__mergeNativeStyleAndNativeClass(undefined, __react__vue__bindStyle(vm['styles'].headerTitle, undefined, undefined))
    }, __react__vue__toString(vm['title'])), vm['subtitle'] ? __react__vue__createElement(vm.$options.components['Text'], {
        style: __react__vue__mergeNativeStyleAndNativeClass(undefined, __react__vue__bindStyle(vm['styles'].headerSubtitle, undefined, undefined))
    }, __react__vue__toString(vm['subtitle'])) : null);
};

const __react__vue__css = {}

const __react__vue__ComponentBuilded = __react__vue__buildNativeComponent(__react__vue__render, __react__vue__options, {Component: __react__vue__Component, PropTypes: __react__vue__PropType, Vue: __react__vue__Vue, ReactNative: __react__vue__ReactNative, css: __react__vue__css})

export default __react__vue__observer(__react__vue__ComponentBuilded) />
`;
